akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -2024,6 +2024,47 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def prefixAndTail(n: Int): javadsl.Flow[In, akka.japi.Pair[java.util.List[Out], javadsl.Source[Out, NotUsed]], Mat] =
     new Flow(delegate.prefixAndTail(n).map { case (taken, tail) => akka.japi.Pair(taken.asJava, tail.asJava) })
 
+  /**
+   * Takes up to `n` elements from the stream (less than `n` only if the upstream completes before emitting `n` elements),
+   * then apply f on these elements in order to obtain a flow, this flow is then materialized and the rest of the input is processed by this flow (similar to via ).
+   * This method returns a flow consuming the rest of the stream producing the materialized flow's output.
+   *
+   * '''Emits when''' the materialized flow emits.
+   *  Notice the first `n` elements are buffered internally before materializing the flow, This flow will then be materialized and connected to the rest of the upstream - producing elements at its own discretion (might 'swallow' or multiply elements).
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' the materialized flow completes.
+   *  If upstream completes before producing `n` elements, `f` will be applied with the provided elements,
+   *  the resulting flow will be materialized and signalled for upstream completion, it can then cancel/complete at its own discretion.
+   *
+   * '''Cancels when''' upstream cancels, application of `f` fails, materialization of the resulting flow fails or the materialized flow cancels.
```

Comment:
```suggestion
   * '''Cancels when''' downstream cancels before the prefix was collected, application of `f` fails, materialization of the resulting flow fails or the materialized flow cancels.
```

Reply:
@jrudolph , this sentence will become obsolete once I implement your major comment about downstream cancellation prior to prefix completion.
I guess it should say something like:

'''Cancels when''' the materialized flow cancels or when failing to materialize the flow.

we also need to add a description of the way downstream cancellation is handled before the prefix is 'ready' (we've agreed on buffering the cause and feeding it to the materialized flow once materialized)/
should this description be part of the 'cancels when' section or a separate section?

Reply:
Let's put it all into this section. In any case, it still needs to be fixed.